crowbar_framework/lib/crowbar/logger/formatter.rb

```diff
@@ -18,20 +18,40 @@ module Crowbar
   module Logger
     class Formatter < ::Logger::Formatter
       include ActiveSupport::TaggedLogging::Formatter
+      Format = "%s, [%s#%d] %5s -- %s\n"
+      ThreadFormat = "%s, [%s#%d:%s] %5s -- %s\n"
 
-      def call(severity, timestamp, progname, msg)
-        final_message = if msg.is_a? String
+      def call(severity, time, progname, msg)
+        if (ENV["CROWBAR_THREADS"].to_i || 10) > 1
+          thread_id = Thread.current.inspect.gsub(/^#<Thread:([^ ]*) .*/, "\\1")
+          ThreadFormat % [severity[0..0], format_datetime(time), $$, thread_id, severity,
+            msg2str(msg)]
+        else
+          Format % [severity[0..0], format_datetime(time), $$, severity,
+            msg2str(msg)]
```

Comment:
Style/AlignArray: Align the elements of an array literal if they span more than one line. (https://github.com/bbatsov/ruby-style-guide#align-multiline-arrays)